src/implementations/vanilla/src/basics/text-link/tests/tests-text-link.html

```diff
@@ -0,0 +1,85 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <link rel="stylesheet" href="../../../../dist/hig.css">
+        <script type="text/javascript" src="../../../../dist/hig.js"></script>
+        <style>
+            .tests-text-link {
+                padding: 40px;
+                margin: 0;
+                display: flex;
+                justify-content: center;
+            }
+        </style>
+    </head>
+    <body>
+        <div class='tests-text-link tests-text-link-defaults'></div>
+        <div class='tests-text-link tests-text-link-with-href'></div>
+        <div class='tests-text-link tests-text-link-secondary'></div>
+        <div class='tests-text-link tests-text-link-secondary-with-href'></div>
+        <div class='tests-text-link tests-text-link-with-onclick'></div>
+        <div class='tests-text-link tests-text-link-with-onclick-and-href'></div>
+        <script>
+
+            // ----------------------------
+            // DEFAULT TEXTLINK
+            // ----------------------------
+
+            var textLinkWithDefaults = new Hig.TextLink();
+            textLinkWithDefaults.mount('.tests-text-link-defaults');
+
+            // ----------------------------
+            // TEXTLINK WITH LINK
+            // ----------------------------
+
+            var textLink = new Hig.TextLink({
+                href: 'https://github.com/Autodesk/hig',
+                text: 'This is a TextLink to HIG repository url'
+            });
+            textLink.mount('.tests-text-link-with-href');
+
+            // ----------------------------
+            // SECONDARY TEXTLINK
+            // ----------------------------
+
+            var secondaryTextLink = new Hig.TextLink({
+                text: 'This is a secondary TextLink',
+                type: 'secondary'
+            });
+            secondaryTextLink.mount('.tests-text-link-secondary');
+
+            // ----------------------------
+            // SECONDARY TEXTLINK WITH HREF
```

Comment:
Would be great to see a TextLink created with the setter functions in the test. It feels a bit redundant, but this way we know that everything is working as expected. 

If you take a look at test-button.html you will see a test written with default properties passed in and another test using the setter methods.

Reply:
Sure.  My thinking was that these are set by way of the setters too (see TextLink#_componentDidMount) but it can't hurt to be more explicit

Reply:
yeah. I saw that pattern.  I think what we had been following the pattern where defaults are passed straight through to the templates, and then we can also use setters. Do you have thoughts @nfiniteset ?

Reply:
Just chatted with @nfiniteset. He said he actually likes the pattern with _componendDidMount better